lib/go-atscfg/meta.go

```diff
@@ -356,29 +359,32 @@ func filterConfigFileDSes(server *Server, deliveryServices []DeliveryService, de
 				warnings = append(warnings, "got deliveryservice with nil id, skipping!")
 				continue
 			}
-			if ds.XMLID == nil {
+			if &ds.XMLID == nil {
 				warnings = append(warnings, "got deliveryservice with nil xmlId (name), skipping!")
 				continue
 			}
 			if _, ok := dssMap[*ds.ID]; !ok && ds.Topology == nil {
 				continue
 			}
-			dses[tc.DeliveryServiceName(*ds.XMLID)] = ds
+			dses[tc.DeliveryServiceName(ds.XMLID)] = ds
 		}
 	} else {
 		for _, ds := range deliveryServices {
 			if ds.ID == nil {
 				warnings = append(warnings, "got deliveryservice with nil id, skipping!")
 				continue
 			}
-			if ds.XMLID == nil {
+			if &ds.XMLID == nil {
```

Comment:
This condition should be `ds.XMLID == ""`

Reply:
done